subsys/shell/shell.c

```diff
@@ -32,6 +32,12 @@
 #define SHELL_MSG_TOO_MANY_ARGS		"Too many arguments in the command.\n"
 #define SHELL_INIT_OPTION_PRINTER	(NULL)
 
+#ifdef CONFIG_SHELL_THREAD_PRIORITY_OVERRIDE
```

Comment:
Just a hint, alternative way to do thing like that is following:
```
#define SHELL_THREAD_PRIORITY \
    COND_CODE_1(CONFIG_SHELL_THREAD_PRIORITY_OVERRIDE,
            (CONFIG_SHELL_THREAD_PRIORITY), (K_LOWEST_APPLICATION_THREAD_PRIO))
```
It's questionable which is better in this context so i'm far from insisting just wanted to mention that.

Reply:
Thank you. this is interesting. I would try to incorporate it next item. 
Next time I would keep an eye on <sys/util.h>
it seems to have useful functions.

Reply:
anyway I was incorporating other changes. I also incorporated this change too.
Thank you for the feedback.